cmake/modules/platform/os/zos.cmake

```diff
@@ -20,12 +20,17 @@
 #############################################################################
 
 list(APPEND OMR_PLATFORM_DEFINITIONS
-	-DJ9ZOS390
-	-DLONGLONG
 	-D_ALL_SOURCE
-	-D_XOPEN_SOURCE_EXTENDED
-	-DIBM_ATOE
+	-D_OPEN_THREADS=2
 	-D_POSIX_SOURCE
+	-D_XOPEN_SOURCE_EXTENDED
+	-D_ISOC99_SOURCE
+	-DLONGLONG
+	-DJ9ZOS390
+	-DOMR_EBCDIC
```

Comment:
The module option `USE_NATIVE_ENCODING` should select either `-DIBM_ATOE` or `-DOMR_EBCDIC` like https://github.com/eclipse/omr/blob/master/omrmakefiles/rules.zos.mk#L82 does.

Reply:
As far as I gather from this comment from @charliegracie (https://github.com/eclipse/omr/issues/1830#issuecomment-340785443) and my experience implementing this PR is that the only reason `atoe.h` and that whole utility exists is because the project was not previously hosted on Git and files were not properly tagged via `chtag` to specify whether they're ASCII or EBCDIC.

This combined with specifying the `CONVLIT` compiler option runs us into the weird state of ASCII/EBCDIC where we need to do runtime conversions. With Git on z/OS all files are properly filetagged as ASCII on checkout and the existing tools (via auto-conversion) transparently just work.

I too like @charliegracie hope to one day soon eliminate that utility once downstream projects start consuming this PR and building the source by checking out via Git.

Reply:
> I'm hesitant about enabling OMR_EBCDIC by default, as it is largely untested

It is running testing as part of this PR and seems to work fine.

Reply:
THe main thing im worried about is openj9. Is it possible to leave this off be default, but enable it as part of omr CI builds?

Reply:
Fixed as part of 5682d6e and 91f1ee5